sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/AzureMonitorConverter.cs

```diff
@@ -119,19 +119,31 @@ private static MonitorBase GenerateTelemetryData(Activity activity)
 
             if (telemetryType == TelemetryType.Request)
             {
-                var url = activity.Kind == ActivityKind.Server ? UrlHelper.GetUrl(partBTags) : GetMessagingUrl(partBTags);
-                var statusCode = GetHttpStatusCode(partBTags);
-                var success = GetSuccessFromHttpStatusCode(statusCode);
-                var request = new RequestData(2, activity.Context.SpanId.ToHexString(), activity.Duration.ToString("c", CultureInfo.InvariantCulture), success, statusCode)
+                string source = null;
+                string statusCode = string.Empty;
+                string url = null;
+                bool success = true;
+
+                switch (activityType)
+                {
+                    case PartBType.Http:
+                        url = activity.Kind == ActivityKind.Server ? HttpHelper.GetUrl(partBTags) : AzureMonitorConverter.GetMessagingUrl(partBTags);
+                        statusCode = HttpHelper.GetHttpStatusCode(partBTags);
+                        success = HttpHelper.GetSuccessFromHttpStatusCode(statusCode);
+                        break;
+                    case PartBType.Azure:
+                        ComponentHelper.ExtractComponentProperties(partBTags, activity.Kind, out _, out source);
+                        break;
+                }
+
+                RequestData request = new RequestData(2, activity.Context.SpanId.ToHexString(), activity.Duration.ToString("c", CultureInfo.InvariantCulture), success, statusCode)
```

Comment:
minor, either add a comment, or extract the magic constant `2`.

Reply:
Could you please explain what does extract the magic constant means?

Reply:
The `RequestData(2, ...` part - it is not clear from this piece of code what does `2` mean.
We probably want to make it easier and there are two options:
1. const
```csharp
new RequestData(VERSION2, ...)
```
2. comment
```csharp
new RequestData(2 /* the ingestion schema version */, ...)
```